### PR TITLE
Tagged connections for telemetry

### PR DESCRIPTION
**Pull request by johngiffin**

### CODE REVIEW

File: lib/coap/socket_server.ex [hunk 1]
```diff
@@ -118,7 +112,7 @@ defmodule CoAP.SocketServer do
   @doc """
     Deliver messages to be sent to a peer
   """
-  def handle_info({:deliver, message, {host, port} = _peer}, %{socket: socket} = state) do
+  def handle_info({:deliver, message, {host, port} = _peer, tag}, %{socket: socket} = state) do
```
Comment (johngiffin) [inline lib/coap/socket_server.ex:115]: This is a bit awkward because connection invokes this and message encoding (needed to determine size) is done here.

Comment (tpitale) [reply lib/coap/socket_server.ex:115]: I think it’s fine, no great option.

**Review by tpitale — APPROVED** [review]
Looks good so far

**Review by tpitale — APPROVED** [review]

File: lib/coap/socket_server.ex [hunk 2]
```diff
@@ -128,7 +122,7 @@ defmodule CoAP.SocketServer do
     :telemetry.execute(
       [:coap_ex, :connection, :data_sent],
       %{size: byte_size(data)},
-      %{host: ip, port: port}
+      %{host: ip, port: port, tag: tag}
```
Comment (tpitale) [inline lib/coap/socket_server.ex]: @johngiffin You mentioned maybe adding the `token`, too. Is that still in the plan?

**Review by tpitale — APPROVED** [review]
🚢

**Pull request merged**